File: Blueprints/BlueprintDefinitions/Msbuild-NETCore_2_1/StepFunctionsHelloWorld-FSharp/template/src/BlueprintBaseName.1/StepFunctionTasks.fs
namespace BlueprintBaseName._1

open System

open Amazon.Lambda.Core

// Assembly attribute to enable the Lambda function's JSON input to be converted into a .NET class.
[<assembly: LambdaSerializer(typeof<Amazon.Lambda.Serialization.Json.JsonSerializer>)>]
()

module StepFunctionTasks =

    type State() =        

        member val Name = null with get, set
        member val Message = null with get, set
        member val WaitInSeconds = 0 with get, set

    let Greeting (state : State) (context: ILambdaContext) =



        if not(String.IsNullOrEmpty(state.Name)) then 
            state.Message <- sprintf "Hello %s" state.Name
        else
            state.Message <- "Hello"

        context.Logger.LogLine(sprintf "Process Greeting message to be: \"%s\"" state.Message)
        state.WaitInSeconds <- 5
        state

    let Salutations (state : State) (context: ILambdaContext) =
        
        if not(String.IsNullOrEmpty(state.Name)) then 
            state.Message <- sprintf "%s, Goodbye %s" state.Message state.Name
        else
            state.Message <- sprintf "%s, Goodbye" state.Message
            
        context.Logger.LogLine(sprintf "Process Salutations message to be: \"%s\"" state.Message)
        state

File: Blueprints/BlueprintDefinitions/netcore3.1/StepFunctionsHelloWorld/template/src/BlueprintBaseName.1/StepFunctionTasks.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Threading.Tasks;

using Amazon.Lambda.Core;


// Assembly attribute to enable the Lambda function's JSON input to be converted into a .NET class.
[assembly: LambdaSerializer(typeof(Amazon.Lambda.Serialization.Json.JsonSerializer))]

namespace BlueprintBaseName._1
{
    public class StepFunctionTasks
    {
        /// <summary>
        /// Default constructor that Lambda will invoke.
        /// </summary>
        public StepFunctionTasks()
        {
        }


        public State Greeting(State state, ILambdaContext context)
        {
            state.Message = "Hello";

            if(!string.IsNullOrEmpty(state.Name))
            {
                state.Message += " " + state.Name;
            }

            // Tell Step Function to wait 5 seconds before calling 
            state.WaitInSeconds = 5;

            return state;
        }

        public State Salutations(State state, ILambdaContext context)
        {
            state.Message += ", Goodbye";

            if (!string.IsNullOrEmpty(state.Name))
            {
                state.Message += " " + state.Name;
            }

            return state;
        }
    }
}
